### ISSUE TITLE
Add linter rule for detecting useless filter anchors at start/end of network rule patterns

### ISSUE THREAD

**Issue by scripthunter7** [AdguardTeam/AGLint#265]
Certain network filter patterns contain meaningless or redundant anchors that do not affect matching and should be removed. Two cases should be detected:

### 1. Useless filter start

Patterns starting with `|*`, e.g.:

```adblock
|*foo
```

This combination is semantically useless (`|` anchors to the start, `*` matches anything).

### 2. Useless filter end

Patterns ending with sequences like `*^`, `^`, `^|`, or `*|`, e.g.:

```adblock
foo^
foo^|
*|
***^
```

These endings provide no meaningful restriction and are effectively redundant.

We should introduce a linter rule that:
- detects useless start/end constructs
- reports them as warnings,
- provides autofix by removing the redundant part of the pattern.
